### PR TITLE
[MERGE] website(_*): reorganize the JS

### PR DESCRIPTION
**Pull request by qsm-odoo**
Reorganize the javascript to use conventional website widgets, rpc calls, classes, ... This will allow to better improve, fix and extend the existing behaviors. This is also a first step towards automatically testing website features more correctly.

This work was started with  https://github.com/odoo/odoo/pull/27719

### CODE REVIEW

File: addons/website_links/static/src/js/website_links_charts.js [hunk 1]
```diff
@@ -1,238 +1,197 @@
...
                 // Process all time line chart data
                 var now = moment();
-
-                charts.all_time_bar = new BarChart('#all_time_clicks_chart', begin_date, now, formatted_clicks_by_day);
+                this.charts.all_time_bar = new BarChart('#all_time_clicks_chart', begin_date, now, formatted_clicks_by_day);
 
                 // Process month line chart data
                 begin_date = moment().subtract(30, 'days');
-                charts.last_month_bar = new BarChart('#last_month_clicks_chart', begin_date, now, formatted_clicks_by_day);
+                this.charts.last_month_bar = new BarChart('#last_month_clicks_chart', begin_date, now, formatted_clicks_by_day);
```
Comment (qsm-odoo) [inline addons/website_links/static/src/js/website_links_charts.js]: @kea14 On your next tasks, please be careful with `this`. Here you cannot use it, this is a reference to the anonymous function you are in. Here, what you wanted to do is define `var self = this;` outside of the function then use `self` in it.

**Comment by qsm-odoo** [comment]
@JKE-be I went over all the files, bug fixed what I saw was wrong (see the "review" commit) and runbot is now green... but I did not test a thing and it would take me a very long time to check if everything was done correctly just by reading the diff. Should I take the time to test myself if no feature was lost or should I ask India to have a look ? Same for the enterprise PR.

**Comment by JKE-be** [comment]
robodoo r+

**Comment by d-fence** [comment]
I think that it's not the job of the mergebot :-)

**Comment by qsm-odoo** [comment]
@d-fence @robodoo creates the message... but ok :no_mouth:

**Comment by d-fence** [comment]
Sorry, you are right. I thought you wanted to verify user commit messages.

**Comment by d-fence** [comment]
But the mergebot will not reformat the massage. Because it's not a good idea.
The message come from PR description and you can reformat it yourself to enforce 72 character length.

**Comment by qsm-odoo** [comment]
@d-fence Mmh, I guess I will have to open a commit file in my editor to write and c/c :/ Just have to remember it. Maybe a simple check like "Your PR description is more than 72 character long, are you sure you want to merge ?".

**Pull request closed without merging**